Kuadrant/multicluster-gateway-controller
# Add Google DNS provider

closes #258 

ToDo

- [x] Load google credentials from secret
- [x] Implement provider.Ensure (DNSRecord create/update)
  ~~- [x] Add mapping for geo location codes to google regions (NOT needed anymore)~~
- [x] Implement provider.Delete(DNSRecord delete)
- [x] Implement providerEnsureManagedZone
- [x] Implement provider.DeleteManagedZone
- [x] Add documentation on how to work with the google provider
~~- [ ] Update e2e tests to include a google managed zone~~ Moving to separate pr



# Verification 
PREREQS (Youll need GCP access)

to be able to curl use ZONE_ROOT_DOMAIN_GCP=google.hcpapps.net
1. Run through the gcp provider setup in the provider doc https://github.com/Kuadrant/multicluster-gateway-controller/pull/340/files#diff-b2417526f682b86f015e23a3ea7a387ed4c95171f0958d8498c5cdce229307caR78
2. run the following to bring up a cluster with gcp: 
make local-setup OCM_SINGLE=true GCP=true
3.  Run through https://github.com/Kuadrant/multicluster-gateway-controller/blob/943e1753342ff106b4471f4ff7884c2532ac7f43/docs/how-to/ocm-control-plane-walkthrough.md (Note quick start doesn't use gcp)
4. Records should be created in GCP with whatever changes eg geo or weighted
 ![Screenshot 2023-08-02 at 15 50 42](https://github.com/Kuadrant/multicluster-gateway-controller/assets/53817495/2f06eba3-5d4e-4708-9ebd-cd49667b538a)
5. Deleting the DNS record cr will delete the records in gcp
